src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -170,21 +166,30 @@ CATCH_RETURN()
         UINT32 uiLengthRead = 0;
 
         // Start from the beginning.
-        const UINT32 glyphStart = 0;
+        UINT32 pos = 0;
 
         _glyphIndices.resize(textLength);
 
-        const HRESULT hr = _fontRenderData->Analyzer()->GetTextComplexity(
-            _text.c_str(),
-            textLength,
-            _fontInUse,
-            &isTextSimple,
-            &uiLengthRead,
-            &_glyphIndices.at(glyphStart));
-
-        RETURN_IF_FAILED(hr);
-
-        _isEntireTextSimple = isTextSimple && uiLengthRead == textLength;
+        while (pos < textLength)
+        {
+            const HRESULT hr = _fontRenderData->Analyzer()->GetTextComplexity(
+                _text.c_str() + pos,

```

Comment:
This is awkward. Auditing complains about pointer arithmetic operations. But DX requires raw pointers as parameter. What should I do?

Reply:
@DHowett and I noodled on this and agreed on `&_text.at(pos)` as something that should satisfy it.